.github/workflows/e2e-test-train-api.yaml

```
@@ -0,0 +1,54 @@
+name: E2E Test with train API
+on:
+  - pull_request
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  e2e-test:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        kubernetes-version: ["v1.28.7"]
+        python-version: ["3.9", "3.10", "3.11"]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Free-Up Disk Space
+        uses: ./.github/workflows/free-up-disk-space
+
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Create k8s Kind Cluster
+        uses: helm/kind-action@v1.10.0
+        with:
+          node_image: kindest/node:${{ matrix.kubernetes-version }}
+          cluster_name: training-operator-cluster
+          kubectl_version: ${{ matrix.kubernetes-version }}
+
+      - name: Build training-operator
+        run: |
+          ./scripts/gha/build-image.sh
+        env:
+          TRAINING_CI_IMAGE: kubeflowtraining/training-operator:test
+
+      - name: Deploy training operator
+        run: |
+          ./scripts/gha/setup-training-operator.sh
+        env:
+          KIND_CLUSTER: training-operator-cluster
+          TRAINING_CI_IMAGE: kubeflowtraining/training-operator:test
+          GANG_SCHEDULER_NAME: "none"
+          KUBERNETES_VERSION: ${{ matrix.kubernetes-version }}
+
+      - name: Run tests
+        run: |
+          pip install pytest
+          python3 -m pip install -e sdk/python -r sdk/python/kubeflow/trainer/requirements.txt; pytest -s sdk/python/test/e2e-train-api/test_e2e_train_api.py --log-cli-level=debug
```

I think, you should install SDK with `huggingface` dependency here.
```suggestion
          python3 -m pip install -e "sdk/python[huggigface]"; pytest -s sdk/python/test/e2e-train-api/test_e2e_train_api.py --log-cli-level=debug
```

---

@andreyvelich We've previously discussed this here: https://github.com/kubeflow/training-operator/pull/2199#discussion_r1737845080. Which way do you think would be better?

---

I see, I think we should install SDK in edit mode with `huggingface` dependency since it verified the correct SDK packages: https://github.com/kubeflow/training-operator/blob/master/sdk/python/setup.py#L67
@tenzen-y Any concerns with that ?

---

We just call this extras as `huggingface`. E.g. `pip install kubeflow-training[huggingface]`

---

@andreyvelich In that case, shouldn't we install the SDK with an editable mode like "pip install -e sdk/python/kubeflow/trainer" so that we can avoid double management of the necessary dependencies?

---

Actually, `trainer` doesn't have any packages to install. 
As you can see Trainer is just a [container that we run on every PyTorchJob replica](https://github.com/kubeflow/training-operator/blob/master/sdk/python/kubeflow/trainer/Dockerfile).
This directory contains the actual SDK files: https://github.com/kubeflow/training-operator/tree/master/sdk/python/kubeflow/training, and we install the required dependencies for [Train API here](https://github.com/kubeflow/training-operator/blob/master/sdk/python/setup.py#L67).

In the future, we place Trainer code in the different location, to not confuse users.

---

That makes sense. In that case, let's keep using the "sdk/python[huggigface]" approach.

---

@helenxie-bit Could you apply the changes to this PR?

---

Yeah, of course, I will adjust it.